lib/configuration/applyConfiguration.js

```diff
@@ -0,0 +1,111 @@
+const R = require('ramda');
+const { EventEmitter } = require('events');
+const logger = require('../logger');
+const getProxySettings = require('../getProxySettings');
+const applyLoggingOptions = require('./applyLoggingOptions');
+const validateConfig = require('./validateConfig');
+const { normalizeConfig } = require('./normalizeConfig');
+
+const defaultConfig = {
+  http: {},
+  server: null,
+  // emitter: new EventEmitter(), // TODO Merge + preserve prototype?
```

Comment:
Nitpick: When I think about it, I'd appreciate if the comment was more verbose. In case we keep it here for a while, it might get murky when we forget the context. Don't be afraid to overcomment in this case:

```js
const defaultConfig = {
  http: {},
  server: null,
  // TODO Merge + preserve prototype?
  // 
  // If the next line was uncommented, the subsequent merging machinery wouldn't be able to deal with
  // the fact the EventEmitter isn't a plain object. We worked around it by making the code to treat the event emitter
  // as a special case, but a more robust solution would be nicer.
  //
  // emitter: new EventEmitter(),
  ...
```

Just an example, perhaps it could be more brief 😄 I'm sometimes way too verbose with my writing.

In case of weird things, special cases, work arounds, etc., I'm all for writing tomes of wisdom for the next generations of readers into the code. Otherwise the suffering you went through to discover and solve the problem will be forgotten and someone else in the future will have to go through it again.

Reply:
Yes, good call. I've provided detailed description below, where we explicitly set `emitter` after merging with default config. I will mention it here as well.